app/backend/app.py

```diff
@@ -229,10 +244,51 @@ def config():
             "showSemanticRankerOption": current_app.config[CONFIG_SEMANTIC_RANKER_DEPLOYED],
             "showVectorOption": current_app.config[CONFIG_VECTOR_SEARCH_ENABLED],
             "showUserUpload": current_app.config[CONFIG_USER_UPLOAD_ENABLED],
+            "showSpeechInput": current_app.config[CONFIG_SPEECH_INPUT_ENABLED],
+            "showSpeechOutput": current_app.config[CONFIG_SPEECH_OUTPUT_ENABLED],
         }
     )
 
 
+@bp.route("/speech", methods=["POST"])
+async def speech():
+    if not request.is_json:
+        return jsonify({"error": "request must be json"}), 415
+
+    speech_token = current_app.config.get(CONFIG_SPEECH_TOKEN)
+    if speech_token is None or speech_token.expires_on < time.time() + 60:
+        speech_token = await current_app.config[CONFIG_CREDENTIAL].get_token(
+            "https://cognitiveservices.azure.com/.default"
+        )
+        current_app.config[CONFIG_SPEECH_TOKEN] = speech_token
+
+    request_json = await request.get_json()
+    text = request_json["text"]
+    try:
+        auth_token = (
+            "aad#" + current_app.config[CONFIG_SPEECH_SERVICE_ID] + "#" + current_app.config[CONFIG_SPEECH_TOKEN].token
+        )
+        speech_config = SpeechConfig(auth_token=auth_token, region=current_app.config[CONFIG_SPEECH_REGION])
+        speech_config.speech_synthesis_voice_name = current_app.config[CONFIG_SPEECH_VOICE]
+        speech_config.speech_synthesis_output_format = SpeechSynthesisOutputFormat.Audio16Khz32KBitRateMonoMp3
+        synthesizer = SpeechSynthesizer(speech_config=speech_config, audio_config=None)
+        result: SpeechSynthesisResult = synthesizer.speak_text_async(text).get()
+        if result.reason == ResultReason.SynthesizingAudioCompleted:
+            return result.audio_data, 200, {"Content-Type": "audio/mp3"}
+        elif result.reason == ResultReason.Canceled:
+            cancellation_details = result.cancellation_details
+            current_app.logger.error(
+                "Speech synthesis canceled: %s %s", cancellation_details.reason, cancellation_details.error_details
+            )
+            raise Exception("Speech synthesis canceled. Check logs for details.")
```

Comment:
Should we include the reason why in the exception? Not sure if we are still including that in the error page @pamelafox

Reply:
We generally don't include reasons in the user-sent errors unless we think they're user-solvable. Instead we log those out. Seems better security-wise.